pkg/image/importer/importer.go

```diff
@@ -388,11 +393,38 @@ func importRepositoryFromDocker(ctx gocontext.Context, retriever RepositoryRetri
 			importDigest.Err = err
 			continue
 		}
-		importDigest.Image, err = schema1ToImage(m, d)
+
+		if signedManifest, isSchema1 := manifest.(*schema1.SignedManifest); isSchema1 {
+			importDigest.Image, err = schema1ToImage(signedManifest, d)
+		} else if deserializedManifest, isSchema2 := manifest.(*schema2.DeserializedManifest); isSchema2 {
+			imageConfig, err := b.Get(ctx, deserializedManifest.Config.Digest)
+			if err != nil {
+				glog.V(5).Infof("unable to access to image config using digest %q for repository %#v: %#v", d, repository, err)
+				switch {
+				case isDockerError(err, v2.ErrorCodeManifestUnknown):
+					ref := repository.Ref
+					ref.Tag, ref.ID = "", importDigest.Name
+					err = kapierrors.NewNotFound(api.Resource("dockerimage"), ref.Exact())
+				case isDockerError(err, errcode.ErrorCodeUnauthorized):
+					err = kapierrors.NewUnauthorized(fmt.Sprintf("you may not have access to the Docker image %q", repository.Ref.Exact()))
+				case strings.HasSuffix(err.Error(), "no basic auth credentials"):
+					err = kapierrors.NewUnauthorized(fmt.Sprintf("you may not have access to the Docker image %q", repository.Ref.Exact()))
+				}
+				importDigest.Err = err
+				continue
+			}
+
+			importDigest.Image, err = schema2ToImage(deserializedManifest, imageConfig, d)
+		} else {
+			glog.V(5).Infof("unsupported manufest type: %#v", manifest)
+			continue
+		}
```

Comment:
@miminar I tried, but this code uses too much from this function: `BlobStore`, `importDigest.Name` and sets `importDigest.Err` or just starts new iteration.